src/Entity/EntityId.php

```diff
@@ -30,6 +37,68 @@ public function getSerialization() {
 	}
 
 	/**
+	 * Returns an array with 3 elements: the foreign repository name as the first element, the local ID as the last
+	 * element and everything that is in between as the second element.
+	 *
+	 * @param $serialization
+	 * @return array
+	 */
+	public static function splitSerialization( $serialization ) {
+		$parts = explode( ':', $serialization );
+		$localPart = array_pop( $parts );
+		$repoName = array_shift( $parts ) ?: '';
+		$prefixRemainder = implode( ':', $parts );
+
+		return array( $repoName, $prefixRemainder, $localPart );
+	}
+
+	/**
+	 * Concatenates parts of an EntityId serialization with ':'.
+	 *
+	 * @param array $parts
+	 * @return string
+	 */
+	public static function joinSerialization( $parts ) {
+		return implode( ':', array_filter( $parts ) );
+	}
+
+	/**
+	 * Returns the foreign repository name or empty string for local entities.
+	 * @return string
+	 */
+	public function getRepoName() {
+		$parts = self::splitSerialization( $this->serialization );
+
+		return $parts[0];
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getLocalPart() {
```

Comment:
Yea... "core ID"? "actual ID"? "local local ID"?
Or the `bar:Q1337` part could be called the "foreign local ID"? Or "relative ID"?... 

Bah. The two hardest things in programming. Cache invalidation, naming things, and off-by-one errors.